### DISCUSSION TITLE
Introduction of Titan’s RetroPGF Program - A Step Forward in Community Contributions

### DISCUSSION THREAD

**Discussion by MaryTitan** [Titannet-dao/Community#3] [RetroPGF]
📣Hello Titan Community!
💫Today marks an exciting milestone in our journey! We are thrilled to introduce the Titan Retroactive Public Goods Funding (RetroPGF) Program, an initiative deeply rooted in the visionary concept proposed by Ethereum’s co-founder, Vitalik Buterin. Our program draws inspiration from the RetroPGF model and aligns with similar actions initiated by projects like Filecoin.

🔥Titan has always been a front-runner in recognizing and rewarding community contributions. With the Titan RetroPGF Program, we are taking this commitment to a new level. Our goal is to continually incentivize individuals and teams that contribute significantly to the Titan ecosystem.

💡What is RetroPGF? Simply put, it’s a way to retrospectively fund deeds and initiatives that have already shown value to our community. It differs from traditional funding by focusing on proven impacts rather than potential future results. This approach ensures that valuable contributions, often overlooked in standard models, get the recognition and support they deserve.

🚀Our program is not just about funding; it’s about acknowledging the hard work, dedication, and innovative spirit that drive our community forward. By participating in Titan’s RetroPGF, you’re not only contributing to the ecosystem but also paving the way for a more collaborative and thriving Titan future.

💪Whether you are a developer, creator, or enthusiast, your efforts and innovations are vital to our growth. We invite you to join us in this pioneering venture. Share your projects, engage in discussions, and let’s work together to build a stronger, more interconnected Titan community.

🌈Let’s embark on this exciting journey together, fostering innovation and collaboration within the Titan ecosystem!
Looking forward to seeing your brilliant contributions!

Warm regards,
Titan Network

**Comment by q3579338** [comment]
1 按地区给积分，奖励稀有地区
2 奖励独享IP节点
3 测试下载别这么大流量

**Reply by MorpheusTNT** [reply]
1 为什么有这样得建议
2 值得考虑
3 是的，同意增加流量的主动控制功能

**Reply by MaryTitan** [reply]
We will take your suggestions into consideration. Thank you very much.

**Reply by susiloadja** [reply]
issue program

**Comment by fcmfcm1999** [comment]
1. Every region should be equally fair. If we want to reward rare areas, how to define rare areas?
2. Agree with [q3579338](https://github.com/q3579338): If an IP only runs one node, it needs to be additionally rewarded
3. Maybe Titan can develop an app for mobile devices, which can allow for wider user participation

**Reply by MorpheusTNT** [reply]
1.Good question
2.Good suggestion
3.Coming soon

**Reply by MaryTitan** [reply]
We will take your suggestions into consideration. Thank you very much.

**Comment by qjaden** [comment]
1. Fair treatment of regions, but with the option to assign certain weights based on traffic.
2. Enable more edge devices to run nodes, such as Raspberry Pi.

**Reply by MorpheusTNT** [reply]
1. The first step is subsidy incentives, and in the future, it may be market demand-driven + subsidy incentives.
2. Yes, Titan not only supports Raspberry Pi but also supports ultra-lightweight hardware devices, and will soon support smartphones.

**Reply by MaryTitan** [reply]
We will allow multiple edge devices including all kinds of smart phones to run nodes in the second round of testnet, which is estimated to go live in mid of May. Thank you for your advice!

**Comment by yeahwong** [comment]
- **完善的文档：** 确保程序使用和参数配置的文档完善且详尽，包括对功能和用法的详细说明。此外，及时更新文档以确保其与程序的最新版本保持一致，这将有助于提高用户的使用体验。

- **网络受限问题：** 考虑到网络受限的情况，应确保去中心化程度类似于电驴或者BitTorrent的协议，以免使用中心化验证节点导致某些特定国家或地区的参与者无法共享资源。这将确保受限访问的国家或地区的用户也能够方便地参与项目。

- **隐私保护：** 对于参与者的隐私和匿名性，应提供详细的保障方案，并在相关说明中清晰阐述实施细节，以确保参与者的隐私权得到充分保护，也应该引入第三方或DAO监督。

- **以太坊数据可用层的实现：** 考虑将Titan打造成以太坊作为数据可用层 Data Availability Layer（DA）。鉴于以太坊二层技术的蓬勃发展，越来越多的项目开始使用第三方DA，这些项目其中不乏估值颇高。可提供分布式和去中心的热数据是TItan的核心功能，以太坊第三方DA的发展给予了Titan成为以太坊数据可用层的可能性，这值得进一步探讨。

- **严格执行路线图：** 确保严格执行项目的路线图，展现诚信和可靠性，这是确保项目成功的关键。

**Reply by MaryTitan** [reply]
Thank you for you suggestions! We will get back to you once confirmed with the tech team!

**Reply by cha0w** [reply]
支持完善文档和公布项目路线图，太重要了

**Reply by yeahwong** [reply]
有一个疑问：目前的系统设计中，边缘节点都是直接从验证节点下载文件？倘若边缘节点都从验证节点下载数据，这样的设计可能存在经济效益不佳以及去中心化优势受损的问题。若能实现边缘节点之间直接相互传输文件，将有助于减轻验证节点的存储、流量和带宽负担，同时也能更有效地实现去中心化。否则，随着边缘节点数量的增加，验证节点的成本可能会逐渐增大，直到无法承承受。

**Reply by MorpheusTNT** [reply]
下载、网络受限、隐私、路线图是这样设计的，将逐一呈现。
文档方面需要更多的社区助力，比如帮忙发现不足，要尽快建立更新流程，文字工作的志愿者参与。
DA层方面欢迎有更细致的描述，可以单独开贴讨论最短路径实施的可能性。可以参考：https://github.com/Titannet-dao/Community/discussions/4

**Reply by MaryTitan** [reply]
I agree. Actually, edge nodes do not download data from validating nodes. The current process of file downloading is that Level 1 (L1) and some high-quality Level 2 (L2) nodes downloading data from the data source site first. Then, based on the distribution strategy, L1 and L2 nodes distribute the data to other L2 nodes.

**Reply by yeahwong** [reply]
原来L2边缘节点之间可以分发数据呀，那就太好了。

**Comment by zzhucong** [comment]
1 按地区给积分，奖励稀有地区
更应该是按（用户）须求多的地区更多积分，因为这样大家就会在（用户）须求更多的地区建立更多节点，用户体验会更好，每一星期或半个月就应该改变（按实际数据情况更新）一次地区的积分占比。
2 奖励独享IP节点
应该是要的，这样可以区分专业与否，更多奖励长时间在线的，比如7*24小时在线与随机在线，更应该奖励专业、稳定的节点，而且要考虑节点资源是否有冗余，毕竟别人一条宽带只放一个节点，有很好的cpu和内存、宽带，可以服务更多的客户应该支持的。如果所有全职7*24的服务器都不赚钱，那业余的就更不用想了，没有利益别想着大家积极去运行节点，IPFS为什么推出filecoin？就是因为奖励驱动fil，然后带动Ipfs，关于利益参考迅雷等前面仪多cdn挖矿项目为什么做不大。
3 测试下载别这么大流量
应该可以让节点自行设置，可以自由调节10-100%，这样个人节点与职业挖矿都可以跟具自己的能力和实际情况贡献资源。本人还是看好职业挖矿的，他们提供的资源稳定，而且可以加点质压积分（代币）来了强化这种资源稳定保证。

**Reply by yeahwong** [reply]
关于第三点，下载的空间目前可以自行限制，速度好像还不可以，

**Reply by MorpheusTNT** [reply]
1 同意
2 有此计划，补贴+市场真实需求+预判市场的真实需求+不断调整。
3 未来网络，可以自行限制，包括带宽，流量，是否下载。

**Comment by zzhucong** [comment]
1 按地区给积分，奖励稀有地区
更应该是按（用户）须求多的地区更多积分，因为这样大家就会在（用户）须求更多的地区建立更多节点，用户体验会更好，每一星期或半个月就应该改变（按实际数据情况更新）一次地区的积分占比。
2 奖励独享IP节点
应该是要的，这样可以区分专业与否，更多奖励长时间在线的，比如7*24小时在线与随机在线，更应该奖励专业、稳定的节点，而且要考虑节点资源是否有冗余，毕竟别人一条宽带只放一个节点，有很好的cpu和内存、宽带，可以服务更多的客户应该支持的。如果所有全职7*24的服务器都不赚钱，那业余的就更不用想了，没有利益别想着大家积极去运行节点，IPFS为什么推出filecoin？就是因为奖励驱动fil，然后带动Ipfs，关于利益参考迅雷等前面仪多cdn挖矿项目为什么做不大。
3 测试下载别这么大流量
应该可以让节点自行设置，可以自由调节10-100%，这样个人节点与职业挖矿都可以跟具自己的能力和实际情况贡献资源。本人还是看好职业挖矿的，他们提供的资源稳定，而且可以加点质压积分（代币）来了强化这种资源稳定保证。

**Reply by MaryTitan** [reply]
The first and second point you mentioned are both about rewards and are under consideration for improvement. The concept of a "reputation vector" will be introduced in the future, which will be associated with factors such as resource allocation, online duration, and others.
As for point 3, bandwidth throttling is under planning and is expected to be updated next week.

**Comment by NewBee0620** [comment]
1、优化节点管理逻辑，目前一个IP可运行5个节点，但是老节点下线之后，无法上线新节点；
2、优化浏览器页面，尤其是离线节点的查看，不是非常直观，没有离线时间排序，不利于快速查找离线节点；
3、项目路线图尽快公布，让矿工心中有数；

**Reply by MaryTitan** [reply]
1. We will update this feature later.
2. Now you can check the conditions of the nodes in 'Console - Node Management', where you can filter out online nodes to see offline nodes. We will update the features with timeline later.
3. Please find the roadmap on our official website (titannet.io), thank you.

**Comment by yummio** [comment]
1. Optimize the console’s online node display and remove duplicate nodes.
2. Consider increasing the active time for node rewards. The longer the stable online time, the higher the reward coefficient.

**Reply by MaryTitan** [reply]
1. Will update soon;
2. It is considered as stated in our Gitbook (https://titannet.gitbook.io/titan-network-en/huygens-testnet/installation-and-earnings/rules-for-obtaining-node-points)

**Comment by eddielueng** [comment]
建议几点：
1、WIN版目前存在修改储存容量后，后台显示可使用的容量还是2G，需要修复下；
2、安卓版本的绑定身份码经常报错的问题；
3、win和安卓版都存在开始挖矿后，一段时间或几分钟后会自动断链的问题；
4、增加独立IP节点奖励，不然5开和1开奖励没多大变化肯定都5开；划分专属额外奖励或激励很有必要，也能保证独立IP节点的价值；
5、后台增加掉线节点搜索功能，方便能快速定位节点，还有增加节点自命名，也方便管理节点；

**Reply by MaryTitan** [reply]
Thank you for your advice. We will work on these asap!
1. It could be a bug. Please reach out to us on [Discord](https://discord.gg/web3depin) and submit a bug report. 
2. Will be updated soon. Sometimes it is due to your phone version or settings.
3. Should be able to reconnect automatically. If not, please check your internet connection.
4. The rule is as you described.
5. You can check offline nodes on the web side.

**Comment by kingcn1312992** [comment]
> RetroPGF

Hello can you explain how to claim the Titan faucet?

**Comment by kingcn1312992** [comment]
Ya

**Comment by Mimiya20** [comment]
> 📣Hello Titan Community! 💫Today marks an exciting milestone in our journey! We are thrilled to introduce the Titan Retroactive Public Goods Funding (RetroPGF) Program, an initiative deeply rooted in the visionary concept proposed by Ethereum’s co-founder, Vitalik Buterin. Our program draws inspiration from the RetroPGF model and aligns with similar actions initiated by projects like Filecoin.
> 
> 🔥Titan has always been a front-runner in recognizing and rewarding community contributions. With the Titan RetroPGF Program, we are taking this commitment to a new level. Our goal is to continually incentivize individuals and teams that contribute significantly to the Titan ecosystem.
> 
> 💡What is RetroPGF? Simply put, it’s a way to retrospectively fund deeds and initiatives that have already shown value to our community. It differs from traditional funding by focusing on proven impacts rather than potential future results. This approach ensures that valuable contributions, often overlooked in standard models, get the recognition and support they deserve.
> 
> 🚀Our program is not just about funding; it’s about acknowledging the hard work, dedication, and innovative spirit that drive our community forward. By participating in Titan’s RetroPGF, you’re not only contributing to the ecosystem but also paving the way for a more collaborative and thriving Titan future.
> 
> 💪Whether you are a developer, creator, or enthusiast, your efforts and innovations are vital to our growth. We invite you to join us in this pioneering venture. Share your projects, engage in discussions, and let’s work together to build a stronger, more interconnected Titan community.
> 
> 🌈Let’s embark on this exciting journey together, fostering innovation and collaboration within the Titan ecosystem! Looking forward to seeing your brilliant contributions!
> 
> Warm regards, Titan Network

**Comment by gandolfslayer** [comment]
How do I become a developer?

**Comment by endebeny** [comment]
Saya harap pihak Titan network bisa memperbaikin console node yang sudah tidak terpakai dengan menambahkan penghapusan node

**Comment by endebeny** [comment]
![Screenshot_2025-03-11-06-25-00-635_com android chrome](https://github.com/user-attachments/assets/0455b8ce-5228-4ead-824b-fdb7f9eafe6a)

Saya harap pihak Titan network bisa memberikan akses untuk menghapus node yang sudah tidak terpakai/Device offline
